### ISSUE TITLE
Solve the mystery of webpack includes and pathing

### ISSUE THREAD

**Issue by SwampGuzzler** [SwampGuzzler/esri-redux#1]
When I try and include and use an esri proxy with webpack, I cannot include it correctly for whatever reason. It appears to be pathing the requests through the proxy, but it returns the Proxy Itself rather than the layer json I would expect!

<img width="1078" alt="screen shot 2016-10-19 at 4 11 41 pm" src="https://cloud.githubusercontent.com/assets/5713523/19536209/feed08c4-9618-11e6-9c3f-1eb8f505d9e5.png">

Above is the request, with a 200 Status Code. However, below is the response: 

<img width="1008" alt="screen shot 2016-10-19 at 4 11 55 pm" src="https://cloud.githubusercontent.com/assets/5713523/19536249/263f7fce-9619-11e6-92af-b544786ecf82.png">

This is all very mysterious, and I am not sure if I am actually requiring in that proxy properly in the 4.1 version of the esri API: https://github.com/SwampGuzzler/esri-redux/blob/master/src/js/components/MapView.js#L39

**Comment by SwampGuzzler** [comment]
@joe-gz

**Comment by SwampGuzzler** [comment]
@Robert-W !?

**Comment by Robert-W** [comment]
Ahh but this makes sense if you think about it.  Express has no idea how to execute php files so it treats any php file or any file loaded via an http request as a static file. So when you make a request to the gfw api, and 4.1 routes that through the proxy, it treats the original request, aka the gfw api, as a query parameter to getting the static php file.  So the 4.1 API seems to be configured correctly and working properly, but express does not know how to execute the php file and use it as a proxy.  You have a couple of options but none that I have actually tried.

* Use something like `php-express` which allows you to execute php in express, but this may be tricky for production because it would require a php binary to be accessible. However if you can point to a php elsewhere in production it would be fine.
* proxy php requests from node to a php capable server, also hard for deployment unless you have a php server you can use in the production environment
* have nginx or apache sit in front of express and use express to serve static assets
* Use this codebase to generate static files, may require some slight changes to the config, and test on your local apache, so you could build and develop like normal, but load the app at `localhost/projects/appfolder`, and point the code to your local proxy.
* You could develop this in docker and link an apache container with a proxy in it, but then deployment would require a container service in amazon or google
* Last and probably the coolest, and I would totally be willing to contribute to something like this, develop a node.js based proxy that you can use. This would be awesome and would allow you guys to build and deploy node.js based apps running on express. This could allow you guys to do some awesome server side rendering, error handling, proxying, and deployment through heroku or any container service.

**Comment by SwampGuzzler** [comment]
Ahh, I am dumb. This makes perfect sense. We cannot execute php because we are running a server-side Express/Node app, where before we were running an in-browser JS app, correct? WEBPAAACK!!

**Comment by Robert-W** [comment]
Yup you got it, you can let the proxy live outside which is what I did for a little bit. Just developed as much as I can using webpack's dev server and then when I needed to test something that needed it, I loaded the browser at `localhost/projects/gfw-water/dist/` or something like that.

**Comment by Robert-W** [comment]
In essence all a proxy does is read the request as a query param and then make the request and hand the results back to the client. This could be done very easily as in JS and setup as a route in express, so all requests to `/proxy` would include some url and params and then hand the results right back.
